File: hw2/newtons.py
from math import e
from sys import argv

verbose = False
for arg in argv:
  if arg == '-v' or arg == '--v' or arg == '--verbose':
    verbose = True

def newtons(f, fPrime, x, maxIterations, tolerance):
  i = 0
  y = 10.0 * tolerance

  if verbose:
    print("{:<12} {:<24} {:<24}".format("Iteration", "Approx. Root", "Abs. Error"))

  while (abs(y) > tolerance and i < maxIterations):
    y = f(x) / fPrime(x)
    x = x - y
    i += 1
    if verbose:
      print("{:<12} {:<24} {:<24}".format(i, x, abs(y)))
  return x

File: hw3/src/newtons.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

double newtons(double (*f)(double), double (*fPrime)(double), double x, double tolerance, int maxIterations) {
  int i = 0;
  double y = 10.0 * tolerance;

  while (fabs(y) > tolerance && i < maxIterations) {
    y = f(x) / fPrime(x);
    x = x - y;
    i++;
  }

  return x;
}
